packages/core/modules-sdk/src/remote-query/query.ts

```diff
@@ -76,6 +96,13 @@ export class Query {
     }
 
     const config = this.#unwrapQueryConfig(queryOptions)
+    if (Query.traceRemoteQuery) {
+      return await Query.traceRemoteQuery(
+        async () => this.#remoteQuery.query(config, undefined, options),
+        queryOptions
```

Comment:
todo: you might want to pass the config instead of the options here, or both eventually